matrix_client/client.py

```diff
@@ -287,6 +287,9 @@ def _process_state_event(self, state_event, current_room):
             current_room.topic = state_event["content"].get("topic", None)
         elif etype == "m.room.aliases":
             current_room.aliases = state_event["content"].get("aliases", None)
+        elif etype == "m.room.member":
+            member_id = state_event["user_id"]
+            current_room.members[member_id] = User(self.api, member_id)
```

Comment:
I've been rereading the spec now, and as I now understand it (specifically, the last sentence in the note at the end of https://matrix.org/docs/spec/client_server/r0.2.0.html#syncing), it seems that the state the client already has should be updated based on the state events, but not processed as timeline events, since they may have happened at some previous time.